cumulus/primitives/storage-weight-reclaim/src/lib.rs

```diff
@@ -471,6 +475,50 @@ mod tests {
 		})
 	}
 
+	#[test]
+	fn test_block_len_weight_is_not_reclaimed() {
+		let mut test_ext = setup_test_externalities(&[100, 200]);
+
+		test_ext.execute_with(|| {
+			set_current_storage_weight(0);
+			// Benchmarked storage weight: 100
+			let info = DispatchInfo { weight: Weight::from_parts(100, 100), ..Default::default() };
+
+			// Actual proof size is 100
+			let post_info = PostDispatchInfo {
+				actual_weight: Some(Weight::from_parts(50, 100)),
+				pays_fee: Default::default(),
+			};
+
+			assert_ok!(CheckWeight::<Test>::do_pre_dispatch(&info, 200));
+			// Weight should go up by 200 len + 100 proof size weight
+			assert_eq!(get_current_storage_weight().total().proof_size(), 300);
+
+			let pre = StorageWeightReclaim::<Test>(PhantomData)
+				.pre_dispatch(&ALICE, CALL, &info, 200)
+				.unwrap();
+			// Should return `setup_test_externalities` proof recorder value: 100.
+			assert_eq!(pre, Some(100));
+
+			// The `CheckWeight` extension will refund `actual_weight` from `PostDispatchInfo`
+			// we always need to call `post_dispatch` to verify that they interoperate correctly.
+			assert_ok!(CheckWeight::<Test>::post_dispatch(None, &info, &post_info, 200, &Ok(())));
+			// `setup_test_externalities` proof recorder value: 200, so this means the extrinsic
+			// actually used 100 proof size.
+			assert_ok!(StorageWeightReclaim::<Test>::post_dispatch(
+				Some(pre),
+				&info,
+				&post_info,
+				200,
+				&Ok(())
+			));
+
+			// Check block len weight was not reclaimed:
+			// 100 weight + 200 extrinsic len == 300 proof size
+			assert_eq!(get_current_storage_weight().total().proof_size(), 300);
```

Comment:
If I understand this test correctly then no weight gets reclaimed here at all. I think the test would better reflect its name if some storage proof weight was reclaimed (but nothing of the extrinsic length).

Maybe naming like `reclaims_as_expected` and then having clear comments and assertions: benchmarked weight/actual weight/expected reclaim/PoV size would make it clearer for future readers what we are actually trying to test here.

Reply:
Yes, also we should add a len to all the other tests too and call `CheckWeight::pre_dispatch` in them.

Reply:
Done